### PR TITLE
3.1.25

### PR DESCRIPTION
**Pull request by dschuff**

### CODE REVIEW

File: bazel/revisions.bzl [hunk 1]
```diff
@@ -2,6 +2,13 @@
 # DO NOT MODIFY
 
 EMSCRIPTEN_TAGS = {
+    "3.1.25": struct(
+        hash = "ff6babb041d0f31575cc16d15ef82c6222ca99b8",
+        sha_linux = "c5ae6b4525845ea36bde89cbf4e1d03de87a2658862d76c6a53bbf8de7c67ff5",
+        sha_mac = "d2581aaa7207f0d9dd9949247f0706bda8561e805d67aec166ed4f3b39c3a3fa",
+        sha_mac_arm64 = "dbcb76036a09248c2a839872c27b87b6d4ccc81e57add4e2a6f5e560a2c530fc",
+        sha_win = "3a86d98d934456a74ed06388c1487d95a0d5a3f31777636453f22e61d57d7fb1",
```
Comment (tlively) [inline bazel/revisions.bzl:10]: Where would I look to verify that these are the correct hashes?

Comment (dschuff) [reply bazel/revisions.bzl:10]: `hash` is a revision in the emscripten-releases repo. the other shas are created by `scripts/update_bazel_workspace.sh` and AFAIK don't correspond to something trivially verifiable.

Comment (dschuff) [reply bazel/revisions.bzl:10]: (but there are tests for the bazel bits that run on circleCI if you look below. I assume they work but probably only Mitch knows for sure :) )

File: emscripten-releases-tags.json [hunk 2]
```diff
@@ -10,6 +10,8 @@
     "latest-releases-upstream": "latest"
   },
   "releases": {
+    "3.1.25": "ff6babb041d0f31575cc16d15ef82c6222ca99b8",
+    "3.1.25-asserts": "6b19d6a8c30d7b83ba2193625fc12cce9ae0206b",
```
Comment (tlively) [inline emscripten-releases-tags.json:14]: There are more differences than I would expect in the emscripten-releases repo between these two hashes. Why do they get different versions of binaryen, llvm, etc?

**Review by tlively — APPROVED** [review]
Thanks for the explanations!

**Comment by kripken** [comment]
Two of these builds errored on linux, and one on test-suites. The same test errored in different modes in linux and test-suites, which seems to make that more worrying than usual flakes?

* https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket/8798099824649614993/+/u/Emscripten_testsuite__core3_/stdout
* https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket/8798096489863537457/+/u/Emscripten_testsuite__cores_/stdout
* https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket/8798099824956800833/+/u/Emscripten_testsuite__ASan_/stdout

**Comment by sbc100** [comment]
> Two of these builds errored on linux, and one on test-suites. The same test errored in different modes in linux and test-suites, which seems to make that more worrying than usual flakes?
> 
> * https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket/8798099824649614993/+/u/Emscripten_testsuite__core3_/stdout
> * https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket/8798096489863537457/+/u/Emscripten_testsuite__cores_/stdout
> * https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket/8798099824956800833/+/u/Emscripten_testsuite__ASan_/stdout

Those all looks like known flakes to me.  I don't think any of them are new.

**Comment by dschuff** [comment]
> Those all looks like known flakes to me. I don't think any of them are new.

That was my observation as well.
I keep vacillating between "holy crap these flakes are terrible we need to fix them right now" and "well they really don't affect us too badly day-to-day, and they only appear on Chromium CI, so maybe it's not so bad?"

**Pull request merged**